src/v/cluster/tx_gateway_frontend.h

```diff
@@ -268,10 +268,12 @@ class tx_gateway_frontend final
     void expire_old_txs();
     ss::future<> expire_old_txs(ss::shared_ptr<tm_stm>);
     ss::future<> expire_old_tx(ss::shared_ptr<tm_stm>, kafka::transactional_id);
-    ss::future<> do_expire_old_tx(
+    ss::future<tx_errc> do_expire_old_tx(
       ss::shared_ptr<tm_stm>,
+      model::term_id term,
       kafka::transactional_id,
-      model::timeout_clock::duration);
+      model::timeout_clock::duration,
+      bool ignore_update_ts);
```

Comment:
this commit mentions a pure refactoring bu there two fields are added to `do_expire_old_tx` can you add a comment what are they meant for ?

Reply:
term used to be acquired inside `do_expire_old_tx` and now this responsibility is shifted to the caller; with ignore_update_ts you're right